src/map/utils/petutils.cpp

```diff
@@ -1418,6 +1489,46 @@ namespace petutils
         PMaster->PPet = PPet;
     }
 
+    CTrustEntity* LoadTrust(CCharEntity* PMaster, uint32 TrustID)
+    {
+        DSP_DEBUG_BREAK_IF(TrustID >= g_PPetList.size());
+        Pet_t* PPetData = g_PPetList.at(TrustID);
+        CTrustEntity* PTrust = new CTrustEntity(PMaster);
+        PTrust->loc = PMaster->loc;
+        PTrust->m_OwnerID.id = PMaster->id;
+        PTrust->m_OwnerID.targid = PMaster->targid;
+
+        // spawn me randomly around master
+        PTrust->loc.p = nearPosition(PMaster->loc.p, CPetController::PetRoamDistance, (float)M_PI);
+        Pet_t* trust = g_PPetList.at(TrustID);
+        PTrust->look = trust->look;
+        PTrust->name = trust->name;
+        PTrust->m_name_prefix = trust->name_prefix;
+        PTrust->m_Family = trust->m_Family;
+        PTrust->m_MobSkillList = trust->m_MobSkillList;
+        PTrust->SetMJob(trust->mJob);
+        PTrust->SetSJob(trust->mJob); // TODO: This may not be true for some trusts
+        PTrust->m_Element = trust->m_Element;
+        PTrust->m_PetID = TrustID;
+        PTrust->status = STATUS_NORMAL;
+        PTrust->m_ModelSize = trust->size;
+        PTrust->m_EcoSystem = trust->EcoSystem;
+
+        // assume level matches master
+        PTrust->SetMLevel(PMaster->GetMLevel());
+        PTrust->SetSLevel(PMaster->GetSLevel());
+
+        //set C magic evasion
+        PTrust->setModifier(Mod::MEVA, battleutils::GetMaxSkill(SKILL_ELE, JOB_RDM, PTrust->GetMLevel()));
+        PTrust->health.tp = 0;
+        PTrust->UpdateHealth();
+        PTrust->health.hp = PTrust->GetMaxHP();
+        PTrust->health.mp = PTrust->GetMaxMP();
+
+        // TODO: Load stats from script
```

Comment:
im not really sure whether to handle these in scripts, or db like regular mobs
@teschnei @TeoTwawki watdo

Reply:
probably the db, I think this is just copy/pasted from the regular pet code which I guess is good enough for now

Reply:
At the moment it's just there to get things to work. It's likely to change in the next PR.

Reply:
I was going to do it in db a table.